### ISSUE TITLE
VB.NET Support

### ISSUE THREAD

**Issue by gumbarros** [WebFormsCore/WebFormsCore#1]
Do the framework supports VB.NET? We have a very large legacy system with WebForms VB.NET, looking forward to use this or https://github.com/twsouthwick/systemweb-adapters-ui.

**Comment by GerardSmit** [comment]
At the moment: yes and no.

I did a small test with Visual Basic since I don't use VB.NET personally (https://github.com/WebFormsCore/WebFormsCore/tree/main/examples/WebForms.VisualBasic.Example).

I found out that source generators are expecting to generate the source for the current language. So it's not possible to mix Visual Basic and C#. Because of this I would have to make two source generators; one for Visual Basic and one for C#.

I've implemented a basic Visual Basic generator that only writes the ID fields. When the application loads, it doesn't find the pre-compiled view and starts generating the view in C#. While this works for basic pages/controls but it won't work with pages/controls that contain Visual Basic code (since we're compiling it as C#). So it's practically useless.

However, Visual Basic is quite big in the WebForms area so it's not a bad idea to implement this. I'll add it to the scope 👍 

---

Small side-note, it doesn't impact your system but it's new in WebForms Core:

For the ViewState generator I'm using `ref struct` (`ViewStateWriter` and `ViewStateReader`) so I can use `System.Span` in the writing and reading. When I try to override the method that writes and reads the viewstate, I get an error that this isn't supported:

```
Error BC30668 : 'ViewStateReader' is obsolete: 'Types with embedded references are not supported in this version of your compiler.'.
```

Meaning I can't implement the new ViewStateAttribute:
```vb
<ViewState> Dim _couter As Integer
```

This isn't a big deal, since you still can use `ViewState`:

```vb
Private Property Counter As Integer
    Get
        Return CInt(ViewState("Counter"))
    End Get
    Set
        ViewState("Counter") = Value
    End Set
End Property
```
